sjk-stacktrace/src/main/java/org/gridkit/jvmtool/stacktrace/StackFrame.java

```diff
@@ -216,6 +215,9 @@ public CharSequence subSequence(int start, int end) {
 
     @Override
     public int hashCode() {
+    	if(hash == 0) {
```

Comment:
It's a [non-benign data race](https://shipilev.net/blog/2016/close-encounters-of-jmm-kind/#wishful-benign-is-resilient). Should be:
```java
int h = hash;
if (h == 0) {
    hash = h = toString().hashCode();
}
return h;
```

and wrong formatting..